string/types.ts

```diff
@@ -0,0 +1,2 @@
+export type TextLocation = { offset: number; line: number; column: number };
+export type TextWithLocation = [text: string, location: TextLocation];
```

Comment:
Since these are so closely connected to the functions, I think it's better to place them in that file with the functions. A standalone `types.ts` file is meant for shared types, as in if we had multiple functions within this module which returned these types or used them as inputs.

Once we have a `CONTRIBUTING.md` (#10) it would be good to locate guidance like this within the file.